packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -497,28 +497,18 @@ describe('KeyringController', () => {
         describe('when there is no existing vault', () => {
           it('should create new vault, mnemonic and keychain', async () => {
             await withController(
-              { cacheEncryptionKey },
-              async ({ controller, initialState, encryptor }) => {
-                const cleanKeyringController = new KeyringController({
-                  messenger: buildKeyringControllerMessenger(),
-                  cacheEncryptionKey,
-                  encryptor,
-                });
-                const initialSeedWord = await controller.exportSeedPhrase(
-                  password,
-                );
-                await cleanKeyringController.createNewVaultAndKeychain(
+              { cacheEncryptionKey, skipVaultCreation: true },
+              async ({ controller }) => {
+                await controller.createNewVaultAndKeychain(password);
+
+                const currentSeedPhrase = await controller.exportSeedPhrase(
                   password,
                 );
-                const currentSeedWord =
-                  await cleanKeyringController.exportSeedPhrase(password);
-                expect(initialSeedWord).toBeDefined();
-                expect(initialState).not.toBe(cleanKeyringController.state);
-                expect(currentSeedWord).toBeDefined();
-                expect(initialSeedWord).not.toBe(currentSeedWord);
+
+                expect(currentSeedPhrase.length).toBeGreaterThan(0);
                 expect(
                   isValidHexAddress(
-                    cleanKeyringController.state.keyrings[0].accounts[0] as Hex,
+                    controller.state.keyrings[0].accounts[0] as Hex,
                   ),
                 ).toBe(true);
                 expect(controller.state.vault).toBeDefined();
```

Comment:
Previously this assertion was not working correctly, because it used the wrong controller reference (it should have been `cleanKeyringController`). Now that we're only using one controller instance, this mistake is no longer possible.